primefaces/src/main/java/org/primefaces/model/JPALazyDataModel.java

```diff
@@ -498,4 +514,10 @@ public interface FilterEnricher<T> extends Serializable {
 
         void enrich(Map<String, FilterMeta> filterBy, CriteriaBuilder cb, CriteriaQuery<?> cq, Root<T> root, List<Predicate> predicates);
     }
+
+    @FunctionalInterface
+    public interface AdditionalFilterMeta extends Serializable {
+
+        Map<String, FilterMeta> process();
```

Comment:
I would just do
Collection<FilterMeta>
if you check, we loop over filterBy.values() only